### ISSUE TITLE
Open3D failed to build with error : downloading 'https://github.com/isl-org/Open3D/releases/download/v0.12.0/mkl-include-2020.1-intel_216-win-64.tar.bz2W' failed on windows using MSVC

### ISSUE THREAD

**Issue by spacelg** [isl-org/Open3D#6851]
### Checklist

- [X] I have searched for [similar issues](https://github.com/isl-org/Open3D/issues).
- [X] For Python issues, I have tested with the [latest development wheel](https://www.open3d.org/docs/latest/getting_started.html#development-version-pip).
- [X] I have checked the [release documentation](https://www.open3d.org/docs/release/) and the [latest documentation](https://www.open3d.org/docs/latest/) (for `main` branch).

### Steps to reproduce the issue

Hi, 

Open3D failed to build with error : downloading 'https://github.com/isl-org/Open3D/releases/download/v0.12.0/mkl-include-2020.1-intel_216-win-64.tar.bz2W' failed on windows using MSVC, It can be reproduced on latest commit fcf98ee on main branch. Could you please help look at this issue? Thanks in advance.

**Steps to reproduce the behavior:**
1. git clone https://github.com/isl-org/Open3D C:\gitP\isl-org\Open3D
2. Open a VS2022 x64 cmd
3. cd C:\gitP\isl-org\Open3D && mkdir build_amd64
4. cmake -G "Visual Studio 17 2022" -A x64 -DCMAKE_SYSTEM_VERSION=10.0.22621.0 -DBUILD_EXAMPLES=OFF -DBUILD_SHARED_LIBS=ON -DSTATIC_WINDOWS_RUNTIME=OFF -DBUILD_COMMON_ISPC_ISAS=ON -DBUILD_LIBREALSENSE=ON -DBUILD_WEBRTC=OFF -DBUILD_UNIT_TESTS=ON -DBUILD_CUDA_MODULE=OFF ..
5. pip3 install wheel
6. msbuild /m /p:Platform=x64 /p:Configuration=Release Open3D.sln /t:Rebuild

Error Log:
[Build.log](https://github.com/user-attachments/files/16081325/Build.log)

### Error message

```shell
**Error Message:**
   149>CustomBuild:
         -- [download 0% complete]
         -- Using src='https://anaconda.org/intel/mkl-include/2020.1/download/win-64/mkl-include-2020.1-intel_216.tar.bz2'
         -- [download 100% complete]
         -- [download 0% complete]
         CMake Error at ext_mkl_include-stamp/download-ext_mkl_include.cmake:170 (message):
           Each download failed!
         
   149>CUSTOMBUILD : error : downloading 'https://github.com/isl-org/Open3D/releases/download/v0.12.0/mkl-include-2020.1-intel_216-win-64.tar.bz2W' failed [C:\gitP\isl-org\Open3D\build_amd64\ext_mkl_include.vcxproj]
                   status_code: 22
                   status_string: "HTTP response code said error"
                   log:
                   --- LOG BEGIN ---
                   timeout on name lookup is not supported
             Trying 140.82.116.3:443...
         
           Connected to github.com (140.82.116.3) port 443
         
           schannel: disabled automatic use of client certificate
         
           ALPN: curl offers h2,http/1.1
         
           schannel: connection hostname (github.com) validated against certificate
           name (github.com)
         
           ALPN: server accepted h2
         
           using HTTP/2
         
           [HTTP/2] [1] OPENED stream for
           https://github.com/isl-org/Open3D/releases/download/v0.12.0/mkl-include-2020.1-intel_216-win-64.tar.bz2W
         
         
           [HTTP/2] [1] [:method: GET]
         
           [HTTP/2] [1] [:scheme: https]
         
           [HTTP/2] [1] [:authority: github.com]
         
           [HTTP/2] [1] [:path:
           /isl-org/Open3D/releases/download/v0.12.0/mkl-include-2020.1-intel_216-win-64.tar.bz2W]
         
         
           [HTTP/2] [1] [user-agent: curl/8.4.0]
         
           [HTTP/2] [1] [accept: */*]
         
           GET
           /isl-org/Open3D/releases/download/v0.12.0/mkl-include-2020.1-intel_216-win-64.tar.bz2W
           HTTP/2
         
           Host: github.com
         
           User-Agent: curl/8.4.0
         
           Accept: */*
         
           
         
           schannel: remote party requests renegotiation
         
           schannel: renegotiating SSL/TLS connection
         
           schannel: connection hostname (github.com) validated against certificate
           name (github.com)
         
           schannel: SSL/TLS connection renegotiated
         
           schannel: remote party requests renegotiation
         
           schannel: renegotiating SSL/TLS connection
         
           schannel: connection hostname (github.com) validated against certificate
           name (github.com)
         
           schannel: SSL/TLS connection renegotiated
         
           HTTP/2 404 
         
           server: GitHub.com
         
           date: Wed, 03 Jul 2024 06:35:32 GMT
         
           content-type: text/plain; charset=utf-8
         
           vary: X-PJAX, X-PJAX-Container, Turbo-Visit, Turbo-Frame, Accept-Encoding,
           Accept, X-Requested-With
         
           cache-control: no-cache
         
           strict-transport-security: max-age=31536000; includeSubdomains; preload
         
           x-frame-options: deny
         
           x-content-type-options: nosniff
         
           x-xss-protection: 0
         
           referrer-policy: no-referrer-when-downgrade
         
           content-security-policy: default-src 'none'; base-uri 'self'; connect-src
           'self'; form-action 'self'; img-src 'self' data:; script-src 'self';
           style-src 'unsafe-inline'
         
           content-length: 9
         
           x-github-request-id: E37B:38A09:2349DF7:23E2581:6684F134
```

### Open3D, Python and System information

```markdown
- Operating system: Windows Server 2022 Datacenter
- Python version: PPython 3.11.4
- Open3D version: output from python: `print(open3d.__version__)`
- System architecture: x86
- Compiler version (if built from source): MSVC
```

**Comment by rott-targetfmi** [comment]
I am also running into this same issue after following the build guide. Did you find any solutions?

**Comment by mab0** [comment]
Same error here

**Comment by AlexandreLaborde** [comment]
I was able to overcome this error by removing the the W at the end. I think is just a typo. 
The file in question is "...\Open3D\3rdparty\mkl\mkl.cmake"

After this I also had to update the hash of the file. it appears that the file was updated at some point but the validation hash was not updated

**Comment by ixs-kosuke-maeda** [comment]
I had same error and solve by @AlexandreLaborde 's comment. Thank you!

Perhaps this issue was caused by the second download URL https://anaconda.org/intel/mkl-include/2020.1/download/win-64/mkl-include-2020.1-intel_216.tar.bz2 for MKL being a broken link recently.

**Comment by ssheorey** [comment]
Thanks all for reporting.

Fixed in PR #6859
